Apex_Trigger/MaxLimitTrigger.trigger

```diff
@@ -0,0 +1,19 @@
+trigger MaxLimitTrigger on Student__c (before insert) 
+{
+	if(Trigger.isInsert) 
+    {
+       List<ID> classIds = new List<ID>();
+       for(Student__c std : Trigger.New) 
+       {
+           classIds.add(std.Class__c);
+       }
+       List<Class__c> ClassList = [Select NumberOfStudents__c,MaxSize__c,(Select Id from Students__r) from Class__c where Id IN :classIds];
```

Comment:
where are we using `(Select Id from Students__r)`

Reply:
remove it from query